### PR TITLE
Fire!

### PR DESCRIPTION
**Pull request by lionleaf**

### CODE REVIEW

File: src/DesertLayer.js [hunk 1]
```diff
@@ -308,7 +310,7 @@ DesertLayer.prototype.createSkybox = function(imagePrefix) {
     materialArray.push(new THREE.MeshBasicMaterial({
       map: Loader.loadTexture(imagePrefix + directions[i] + imageSuffix),
       side: THREE.BackSide,
-      transparent: true
+      transparent: false
```
Comment (iver56) [inline src/DesertLayer.js]: This breaks the first skybox. We need have transparent set to true during the transition to the doom skybox, but we can set it to false just before the fire joins the party. Somebody needs to fix this.

File: src/DesertLayer.js [hunk 2]
```diff
@@ -395,6 +397,17 @@ DesertLayer.prototype.initGrass = function() {
...
       clonedObject.scale.set(scale, scale * 1.25, scale * 1.5);
       clonedObject.initScale = scale;
+    
+      if(Math.random() < 0.25){
+        var firePos = clonedObject.position.clone();
+        firePos.y = that.grass.targetY;
+        clonedObject.fire = that.fireFactory.create(firePos, scale, 300, 200);
+        var catch_fire_spread = 700;
+        clonedObject.fire.offset = Math.random() * catch_fire_spread;
+        clonedObject.hasFire = true;
```
Comment (iver56) [inline src/DesertLayer.js:428]: This is ok for now (it's a hackathon, etc) but use the userData object in the future. It is reserved for this kind of stuff.

File: src/DesertLayer.js [hunk 3]
```diff
@@ -773,12 +786,29 @@ DesertLayer.prototype.updateGrass = function(frame, relativeFrame) {
...
     }
   }
-
   //windy
   if (relativeFrame >= this.config.grass.startGrowthFrame
     && relativeFrame < this.config.grass.windyUntilFrame) {
     for (var i = 0; i < this.grasses.length; i++) {
       this.grasses[i].rotation.x = 0.16 * Math.sin(relativeFrame * 0.033 + 1.5 * i / this.grasses.length);
+      if(this.grasses[i].hasFire){
+        this.grasses[i].fire.rotation.x = 0.16 * Math.sin(relativeFrame * 0.033 + 1.5 * i / this.grasses.length);
```
Comment (iver56) [inline src/DesertLayer.js]: this.grasses[i].fire.rotation.x = this.grasses[i].rotation.x

File: src/FireFactory.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ * #### FireFactory     ####
+ * ####      Usage      ####
+ *
+ * ## In constructor of layer ###
+ *
+ * ## In update function ##
+ */
+function FireFactory( scene ) {
+  var firetexture = Loader.loadTexture('res/textures/fireparticle.png');
+  Loader.start(function(){},function(){});
```
Comment (iver56) [inline src/FireFactory.js:12]: Not sure if this is necessary... @stianjensen ?

**Comment by lionleaf** [comment]
screenshot from 2014-07-12 19 56 06

File: src/FireFactory.js (hunk 4 above)
Comment (stianjensen) [inline src/FireFactory.js:12]: It is...

But for compatibility with compiling, it should be written like this:

```
if (!window.FILES) {
  Loader.start(function(){}, function(){});
}
```

File: src/DesertLayer.js (hunk 1 above)
Comment (cristeahub) [inline src/DesertLayer.js]: Another thing is that the rest of the demo has transparent set to true, so we might get unwanted effects

**Pull request merged**